agent/s3/factory/factory.go

```diff
@@ -56,43 +58,55 @@ func (*s3ClientCreator) NewS3ManagerClient(bucket, region string,
 		WithCredentials(
 			awscreds.NewStaticCredentials(creds.AccessKeyID, creds.SecretAccessKey,
 				creds.SessionToken)).WithRegion(region)
+	if config.IsFIPSEnabled() {
+		logger.Debug("FIPS mode detected, using FIPS-compliant S3 endpoint")
+		cfg.UseFIPSEndpoint = endpoints.FIPSEndpointStateEnabled
+	}
 	sess := session.Must(session.NewSession(cfg))
 	svc := s3.New(sess)
 	bucketRegion, err := getRegionFromBucket(svc, bucket)
 	if err != nil {
 		return nil, err
 	}
-	sessWithRegion := session.Must(session.NewSession(cfg.WithRegion(bucketRegion)))
+	cfg.WithRegion(bucketRegion)
+	sessWithRegion := session.Must(session.NewSession(cfg))
 	return s3manager.NewDownloaderWithClient(s3.New(sessWithRegion)), nil
 }
 
-// NewS3Client returns a new S3 client to support s3 operations which are not provided by s3manager.
+// NewS3Client returns a new S3 client to support S3 operations which are not provided by s3manager.
 func (*s3ClientCreator) NewS3Client(bucket, region string,
 	creds credentials.IAMRoleCredentials) (s3client.S3Client, error) {
 	cfg := aws.NewConfig().
 		WithHTTPClient(httpclient.New(roundtripTimeout, false, agentversion.String(), config.OSType)).
 		WithCredentials(
 			awscreds.NewStaticCredentials(creds.AccessKeyID, creds.SecretAccessKey,
 				creds.SessionToken)).WithRegion(region)
+	if config.IsFIPSEnabled() {
+		logger.Debug("FIPS mode detected, using FIPS-compliant S3 endpoint")
+		cfg.UseFIPSEndpoint = endpoints.FIPSEndpointStateEnabled
+	}
 	sess := session.Must(session.NewSession(cfg))
 	svc := s3.New(sess)
 	bucketRegion, err := getRegionFromBucket(svc, bucket)
 	if err != nil {
 		return nil, err
 	}
-	sessWithRegion := session.Must(session.NewSession(cfg.WithRegion(bucketRegion)))
+	cfg.WithRegion(bucketRegion)
```

Comment:
same comment as above

Reply:
Addressed in [f3026bf](https://github.com/aws/amazon-ecs-agent/commit/f3026bff9abc8678c4eefbc5ae07cb1c2eabd283)